docs/api-reference/next.config.js/introduction.md

```diff
@@ -44,7 +44,7 @@ module.exports = (phase, { defaultConfig }) => {
 }
 ```
 
-The commented lines are the place where you can put the configs allowed by `next.config.js`, which are defined [here](https://github.com/vercel/next.js/blob/canary/packages/next/next-server/server/config.ts#L12-L63).
+The commented lines are the place where you can put the configs allowed by `next.config.js`, which are defined [here](https://github.com/vercel/next.js/blob/v10.0.8/packages/next/next-server/server/config-shared.ts#L33).
```

Comment:
It's not likely we'll change the location of the configs very often, but the configs are more likely to change, so it's okay to point it to `canary`

```suggestion
https://github.com/vercel/next.js/blob/canary/packages/next/next-server/server/config-shared.ts#L33
```

Reply:
It might be good to add a comment above this line in the actual code noting we need to update this doc if the location does end up changing in the future.

Reply:
We should document all the options in the docs instead.

Reply:
@lfades Committed your suggestions

> We should document all the options in the docs instead imo.

@timneutkens if we do that, then we have to update configs in both places: docs and code. Wouldn't it be easier to just maintain a link as a pointer to the config object? To @lfades point, the location of the config should rarely change, but properties on the config might; therefore a link to the config is more "future proof".